content/blog/blog-2020-02-07-kudo-terraform-provider-1.md

```diff
@@ -0,0 +1,783 @@
+---
+date: 2020-02-07
+---
+
+# KUDO Terraform Provider
+
+[KUDO](https://kudo.dev) is rolling out new and amazing features every month, but there still are plenty of features that are needed to facilitate usage for production workflows.  This KUDO Terraform provider looks to provide a solution for managing interconnected `Instances` and passing information from one Instance to another in a manner that allows for seemless upgrades.
+
+<!-- more -->
+
+## Current Process
+
+KUDO has the ability to provide `Instance`s specific values to the application through the use of `PARAMETERS`.  These parameters allow for `strings` to be used as inputs to the template rendering built into KUDO.  More about the use of Parameters can be found [Here](https://kudo.dev/docs/developing-operators/parameters.html).
+
+The following example shows more explicitly how parameters are set and managed in a multi-`Instance` environment. In this simple example, supposed we want to deploy a [Kafka](https://github.com/kudobuilder/operators/tree/master/repository/kafka) cluster backed by a [Zookeeper](https://github.com/kudobuilder/operators/tree/master/repository/zookeeper) cluster defined with the following relationship:   Once complete, applying the following objects to the Kubernetes cluster represent that relationship:
+
+```yaml
+apiVersion: kudo.dev/v1beta1
+kind: Instance
+metadata:
+    name: zook
+    namespace: default
+spec:
+    operatorVersion:
+        name: zookeeper-0.3.0
+        namespace: default
+    parameters:
+        NODE_COUNT: "3"
+---
+apiVersion: kudo.dev/v1beta1
+kind: Instance
+metadata:
+    name: pipes
+    namespace: default
+spec:
+    operatorVersion:
+        name: kafka-1.2.0
+        namespace: default
+    parameters:
+        ZOOKEEPER_URI: zook-zookeeper-0.zook-hs:2181,zook-zookeeper-1.zook-hs:2181,zook-zookeeper-2.zook-hs:2181
+```
+
+The `ZOOKEEPER_URI` value was hand calculated with knowledge of what the `NODE_COUNT` value was set to in the Zookeeper `Instance`. 
+
+Before applying these objects, the cluster would need to be configured 
+Before applying this example to a cluster, the KUDO CLI is needed to [install KUDO](https://kudo.dev/docs/#getting-started) and  [install the `Operators`](https://kudo.dev/docs/#deploy-your-first-operator) needed.
```

Comment:
Nit: That last 'needed' is repetitive, maybe remove it.  Also the use of "Before applying..." to start both sentences is a little unwieldy - I think you can also just remove that first sentence.